aries_vcx/src/protocols/issuance/holder/state_machine.rs

```diff
@@ -176,6 +177,11 @@ impl HolderSM {
                             return Some((uid, message));
                         }
                     }
+                    AriesMessage::Notification(Notification::ProblemReport(msg)) => {
+                        if matches_opt_thread_id!(msg, self.thread_id.as_str()) {
```

Comment:
Slightly out of PR scope, but consider this modification while we are at it:

```
#[derive(Debug, Clone, Serialize, Deserialize, Default, PartialEq)]
pub struct ProblemReportDecorators {
    #[serde(skip_serializing_if = "Option::is_none")]
    #[serde(rename = "~thread")]
    pub thread: Option<Thread>,
```
I am not sure if thread would ever be None. It's not expected to send problem reports out of blue, I think it would always be in relation to some conversation, hence I would expect ~thread decorator to always be present for that type of message.

Reply:
i see, good observation. Yeah I also don't know, anyway not a big deal. Let's consider this resolved